### PR TITLE
Refactor Notifications with panoptes-js talkAPI

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-classifier

## Linked Issue and/or Talk Post
- towards #2459

## Describe your changes
- refactor Notifications initial unread notifications request with `panoptes-js` `talkAPI`, in a similar way to [QuickTalk helpers](https://github.com/zooniverse/front-end-monorepo/tree/master/packages/lib-classifier/src/components/Classifier/components/QuickTalk/helpers)

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
- What user actions should my reviewer step through to review this PR?
- Which storybook stories should be reviewed? - N/A
- Are there plans for follow up PR’s to further fix this bug or develop this feature? - yes, incoming Messages unread count PRs will also utilize `panoptes-js` `talkAPI`

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

**Comment by shaunanoordin** [comment]
Tagging myself in for a review! One second while I spool up my local FEM...

**Review by shaunanoordin — APPROVED** [review]
## PR Review

Package: `lib-classifier`

This PR is a refactor of the Notifications system.

- The old Panoptes Talk client has been replaced with the newer `@zooniverse/panoptes-js`
- Code changes look pretty straightforward 👍 
- ⚠️ No manual testing, though. I was unable to get a baseline test to work properly (my knowledge of Talk behaviour is a bit weak); in the future it might be worth writing the PR creator's recommended test steps and expected functionality/behaviours.

### Status

LGTM 👍

File: packages/app-project/stores/User/UserPersonalization/Notifications/Notifications.js
Comment (mcbouslog) [reply packages/app-project/stores/User/UserPersonalization/Notifications/Notifications.js:35]: great suggestion to copy lib-classifier's getBearerToken helper function, will do!

**Comment by mcbouslog** [comment]
@shaunanoordin let me know your staging username and I can mention you in a Talk comment to trigger a Notification, like:
Screen Shot 2022-06-29 at 4 56 52 PM

**Comment by mcbouslog** [comment]
Subsequent to this PR I'll create an issue or try to refactor other instances of auth and Bearer Token in `app-project`

File: packages/app-project/stores/utils/getBearerToken.js
Comment (eatyourgreens) [inline packages/app-project/stores/utils/getBearerToken.js:11]: The project app doesn’t use PJC’s `oauth` so you don’t need this logic here.

Comment (eatyourgreens) [reply packages/app-project/stores/utils/getBearerToken.js:11]: In the project app, `authToken` is always a string containing the token.

File: packages/app-project/stores/User/UserPersonalization/Notifications/Notifications.js
Comment (eatyourgreens) [reply packages/app-project/stores/User/UserPersonalization/Notifications/Notifications.js]: That function contains a bunch of logic that the project app doesn’t need. The classifier uses two different auth clients, so needs logic to handle both. The project app only uses the `auth` module from PJC, so you can remove the code that handles the `oauth` module.

Comment (eatyourgreens) [reply packages/app-project/stores/User/UserPersonalization/Notifications/Notifications.js]: Having said that, the classifier’s `usePanoptesAuth` hook might be useful here.

**Comment by eatyourgreens** [comment]
I started a refactor of project app auth, using data-fetching hooks, in #2979, so that might be worth a look.

File: packages/app-project/stores/User/UserPersonalization/Notifications/Notifications.js
Comment (mcbouslog) [reply packages/app-project/stores/User/UserPersonalization/Notifications/Notifications.js:35]: I've removed the recent commits related to `getBearerToken` for now.

**Comment by mcbouslog** [comment]
I think I'd like to keep this as is (similar to other instances that create a Bearer token in app-project like in Collections, Recents, UserProjectPreferences, and YourStats). I've got two other related PRs coming subsequent to this one to add the unread count for Messages, and can refactor similar to #2979 as Jim notes once that's merged. I'm not set on that though, open to suggestion!

@shaunanoordin do please let me know your staging username and I can create some notifications for you if you'd like (sorry I dismissed your earlier review).

**Review by shaunanoordin — APPROVED** [review]
I'm back!

## PR Review

Follows [earlier review](https://github.com/zooniverse/front-end-monorepo/pull/3347#pullrequestreview-1021570423)
Package: `app-project` (wait, I mistakenly said lib-classifier in the earlier review, oops)

This PR is a refactor of the Notifications system.

- Code changes are pretty straightforward, and no functionality/behaviour has been changed
- I had [some minor/moderate comments](https://github.com/zooniverse/front-end-monorepo/pull/3347/files#r908365683) about the patterns surrounding `const authorization = Bearer ${token}`, but that has already been discussed and the conversation resolved. 👌 
- Testing works! Finally figured out how to properly get notifications.

### Testing

Tested on localhost with macOS + Chrome

Testing Steps:
- Open any project on app-project [(example)](https://local.zooniverse.org:3000/projects/darkeshard/test-project-2022/classify/workflow/3581?env=staging) and sign in as a user, e.g. `darkeshard`
  - Observe that the Notifications button on the top of the page. This should be "empty" in most cases, with the label only saying `Notifications`
- On a separate (incognito) window/tab, open a Talk page for that project (either on zooniverse.org, or in my case, via QuickTalk), and login as another user, e.g. `zootester1`
  - Post a comment tagging the first logged-in user, `Hello @darkeshard`
- Expected: back on the first window/tab, which has `app-project` running, the Notifications button should update to indicate new messages.
  - Assuming only one new message was posted by `zootster1`, the Notifications label should now read `Notifications (1)`
  - Not sure if you have to refresh the page to see the changes, but I believe didn't have to.

Expectations meet actual reality, so this looks good. 🆗 

### Status

LGTM 👍

**Pull request merged**
